8.x/theme-development/helpers.md

```diff
@@ -2,4 +2,2097 @@
+                        $q1->where('p_structure_attr_key_id', $key_id)
+                            ->whereNotIn('p_structure_attr_value_id', $product_key_values);
+                    })->where(function ($q2) use ($key_id, $product_other_keys_values) {
+                        foreach ($product_other_keys_values as $item) {
+                            $key = $item['key'];
+                            $value = $item['value'];
+                            $q2->whereHas('pAttributes', function ($q3) use ($key_id, $key, $value) {
+                                $q3->where('p_structure_attr_key_id', '!=', $key_id);
+                                $q3->where([
+                                    'p_structure_attr_key_id' => $key,
+                                    'p_structure_attr_value_id' => $value
+                                ]);
+                            });
+                        }
+                    })->orderBy("priority", "ASC")->take($count)->get();
+            }
+        }
+        return $products;
+    }
+}
+```
+
+
+### get_related_products_with_directory_level()
+Returns similar products up to level 1 of a directory.
+
+```php
+if (!function_exists('get_related_products_with_directory_level')) {
+    function get_related_products_with_directory_level(Product $product, int $count = 5, int $level = 1)
+    {
+        $directory = $product->directory;
+        if ($level > 1) {
+            while ($level != 1) {
+                $directory = $directory->parentDirectory;
+                $level--;
+            }
+        }
+        return $directory->leafProducts()->mainModels()->visible()->except($product->id)
+            ->orderBy("priority", "ASC")->take($count)->get();
+    }
+}
+```
+
+### get_product_attributes()
+Returns the attributes of a product.
+
+```php
+if (!function_exists('get_product_attributes')) {
+    function get_product_attributes(Product $product = null)
+    {
+        if ($product != null) {
+            $attributes = ProductAttribute::getProductAttributes($product);
+            return $attributes['attributes'];
+        } else {
+            return null;
+        }
+    }
+}
+```
+
+
+### get_product_most_privileged_key_attributes()
+This function takes a list of all the keys and then returns the validity of that key. This function will be deprecated in the next version.
+
+```php
+if (!function_exists('get_product_most_privileged_key_attributes')) {
+      /**
+     * @deprecated
+     */
+    function get_product_most_privileged_key_attributes(int $count = 9): array
+    {
+        $id = ProductStructureAttributeKey::orderBy('priority', 'DESC')->pluck('id')->first();
+        $attributeValues = ProductStructureAttributeValue::with('key')->where('p_structure_attr_key_id', $id)
+            ->inRandomOrder()
+            ->paginate($count);
+        return $attributeValues->items();
+    }
+}
+```
+
+
+### get_product_By_id()
+This function will be deprecated in the next version.
```

Comment:
Remove this.
If you want to keep it, there must be a description.
This note works for all the deprecated functions.
If you want to mention them, please describe them; it's useless just to say that they will be disappeared.